docs/development/index.rst

```diff
@@ -0,0 +1,26 @@
+Development
+===========
+
+pip, as an open-source project, welcomes contributions of all forms. The
+sections below will help you get started with development, testing, and
+documentation.
+
+Please contribute issues, submit bug reports, file feature requests, and help
```

Comment:
This section, as a whole, could possibly be worded better.

Reply:
One thing I think we should mention here is that we support Python 2.7 and 3, and we support Windows and Unix. Contributions *must* consider all of those cases. While a contribution from someone who only works on Python 2.7 and Unix is appreciated, it's not going to progress unless *someone* puts in the work to consider Python 3 and Windows, and I think we need to set contributors' expectations here. More generally, contributions need to go beyond "works for me". As a team, we don't have the manpower to take incomplete contributions and complete them ourselves (although we will happily help the contributor to do so with advice and suggestions).

Reply:
This is probably something that can be clarified in the contributing page, in the "Submitting Pull Requests" section.

Reply:
On the point of setting contributors' expectations early, totally agree. I think this should mention that pip is a volunteer maintained project.

We could also link from the PR template to the "Submitting Pull Requests" section (while also talking about NEWS fragments).

Reply:
Coming back around to this, I don't even think this paragraph is even needed here. We don't need to list various ways to interact with the project on GitHub in the "development" documentation.